lib/Basics/FileUtils.cpp

```diff
@@ -900,4 +900,38 @@ std::optional<gid_t> findGroup(std::string const& nameOrId) noexcept {
   return {std::nullopt};
 }
 #endif
+
+#ifdef ARANGODB_HAVE_INITGROUPS
+void initGroups(std::string const& userName, gid_t groupId) noexcept {
+#ifdef __linux__
+  // For Linux, calling initgroups poses problems with statically linked
+  // binaries, since /etc/nsswitch.conf can then lead to crashes on
+  // older Linux distributions. Therefore, we need to do the groups lookup
+  // ourselves using the groups command. Then we can use setgroups to
+  // achieve the desired result.
+  try {
+    std::vector<std::string> args{"groups", userName};
```

Comment:
Is it necessary to pass the extra `groups` argument to the `/usr/bin/groups` executable? On my local Linux this does not work, but the user name needs to be passed without additional parameters.
```suggestion
    std::vector<std::string> args{userName};
```